conda/core/solve.py

```diff
@@ -1029,6 +1039,251 @@ def _prepare(self, prepared_specs):
         return self._index, self._r
 
 
+class LibSolvSolver(Solver):
+    """
+    This alternative Solver logic wraps ``libsolv`` through their Python bindings.
+
+    We mainly replace ``Solver.solve_final_state``, quite high in the abstraction
+    tree, which means that ``conda.resolve`` and ``conda.common.logic`` are no longer
+    used here. All those bits are replaced by the logic implemented in ``libsolv`` itself.
+
+    This means that ``libsolv`` governs processes like:
+
+    - Building the SAT clauses
+    - Pruning the repodata (?) - JRG: Not sure if this happens internally or at all.
+    - Prioritizing different aspects of the solver (version, build strings, track_features...)
+    """
+
+    def __init__(self, prefix, channels, subdirs=(), specs_to_add=(), specs_to_remove=(),
+                 repodata_fn=REPODATA_FN, command=NULL):
+        super().__init__(prefix, channels, subdirs=subdirs, specs_to_add=specs_to_add,
+                         specs_to_remove=specs_to_remove, repodata_fn=repodata_fn,
+                         command=command)
+
+    def solve_for_transaction(self, update_modifier=NULL, deps_modifier=NULL, prune=NULL,
+                              ignore_pinned=NULL, force_remove=NULL, force_reinstall=NULL,
+                              should_retry_solve=False):
+        # Logic heavily based on Mamba's implementation (solver parts):
+        # https://github.com/mamba-org/mamba/blob/fe4ecc5061a49c5b400fa7e7390b679e983e8456/mamba/mamba.py#L289
+
+        print("------ USING EXPERIMENTAL MAMBA INTEGRATIONS ------")
+
+        # 1. Populate repos with installed packages
+        state = self._setup_state()
+        # 2. Create solver and needed flags, tasks and jobs
+        self._configure_solver(state)
+        # 3. Run the SAT solver
+        self._run_solver(state)
+        # 5. Export back to conda
+        return self._build_transaction(state)
+
+    def _setup_state(self):
+        from mamba.utils import load_channels, get_installed_jsonfile, init_api_context
```

Comment:
We need to figure out a strategy for this, whether we can actually rely on runtime dependencies on Mamba since it could make maintenance of Conda much harder if API is used that may be less stable than Conda's API.